chainer/distribution.py

```diff
@@ -0,0 +1,260 @@
+import copy
+
+
+class Distribution(object):
+
+    """Interface of Distribution
+
+    `Distribution` is a bass class to treat probability distributions.
+    When initialization, it takes parameter as input.
+    """
+
+    def _copy_to(self, target):
+        target.__dict__ = copy.copy(self.__dict__)
+        return target
+
+    @property
+    def batch_shape(self):
+        """Returns the shape of a sample.
+
+        Returns:
+            ~chainer.Variable: Output variable representing the shape of a
+            sample.
+
+        """
+        raise NotImplementedError
+
+    def cdf(self, x):
+        """Returns Cumulative Distribution Function for a input variable.
+
+        Args:
+            x(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+            :class:`cupy.ndarray`): Input variable representing a random
```

Comment:
Another point: could you write the specification of input/output shapes? For example, can `x` be a batch of multiple examples, or should it represent only one example? Ditto for other methods (if most methods have the same specification, it is also ok to write that in the document of `Distribution` class itself instead of repeating it at every method).